src/proxy.c

```diff
@@ -30,25 +30,21 @@ SEXP vec_proxy(SEXP x) {
 
 // [[ register(); include("vctrs.h") ]]
 SEXP vec_proxy_equal(SEXP x) {
-  return vec_proxy_recursive(x, vctrs_proxy_equal);
+  return vec_proxy_unwrap(vec_proxy_recursive(x, vctrs_proxy_equal));
```

Comment:
Is rchk going to be mad that we aren't protecting the result of `vec_proxy_recursive()`?

Reply:
it would even be right to complain 😬